smartlearner/views.py

```diff
@@ -18,30 +18,34 @@ def update(self, status):
 
 
 class ClassificationError(View):
-    def __init__(self, predict_fct, dataset):
-        super(ClassificationError, self).__init__()
-
-        batch_size = 1024  # Internal buffer
-        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
+    def __init__(self, predict_fct, dataset, batch_size=None):
+        super().__init__()
+        self._batch_size = batch_size
 
         input = dataset.symb_inputs
         target = dataset.symb_targets
         classification_errors = T.neq(predict_fct(input), target)
 
-        no_batch = T.iscalar('no_batch')
-        givens = {input: dataset.inputs[no_batch * batch_size:(no_batch + 1) * batch_size],
-                  target: dataset.targets[no_batch * batch_size:(no_batch + 1) * batch_size]}
-        self.compute_classification_error = theano.function([no_batch],
+        if batch_size is None:
+            inp = []
```

Comment:
If `batch_size is None` I would simply set `self._batch_size = len(dataset)` so that the following code is clearer (no needs for the 'if' in method `update` anymore). Also, I don't think it will have a real impact on the computational speed.